cmd/xl-storage-disk-id-check.go

```diff
@@ -720,7 +720,9 @@ func (p *xlStorageDiskIDCheck) ReadParts(ctx context.Context, volume string, par
 	}
 	defer done(0, &err)
 
-	return p.storage.ReadParts(ctx, volume, partMetaPaths...)
+	return xioutil.WithDeadline[[]*ObjectPartInfo](ctx, globalDriveConfig.GetMaxTimeout(), func(ctx context.Context) ([]*ObjectPartInfo, error) {
```

Comment:
This seems a bit risky, considering we are reading up to 10K files (and stat'ing the same number of files).

Reply:
yeah that is correct, let me change it.